### PR TITLE
Disarm, Silence and Lulu W

### PR DESCRIPTION
**Pull request by Deudly**
Disarm, Silence and Lulu W

### CODE REVIEW

File: Champions/Lulu/W.cs [hunk 1]
```diff
@@ -15,16 +15,59 @@ public class W : GameScript
...
+                ApiFunctionManager.SetChampionModel((Champion)target, "LuluSquill");
+            } else if (skinId == 1){
+                ApiFunctionManager.SetChampionModel((Champion)target, "LuluCupcake");
+            } else if (skinId == 2){
+                ApiFunctionManager.SetChampionModel((Champion)target, "LuluKitty");
+            } else if (skinId == 3){
+                ApiFunctionManager.SetChampionModel((Champion)target, "LuluDragon");
+            } else if (skinId == 4){
+                ApiFunctionManager.SetChampionModel((Champion)target, "LuluSnowman");
+            }
```
Comment (molenzwiebel) [inline Champions/Lulu/W.cs]: Inconsistent indentation.

File: Champions/Lulu/W.cs [hunk 2]
```diff
@@ -15,16 +15,59 @@ public class W : GameScript
...
-
-        }
+            spell.spellAnimation("SPELL2", owner);
+         }
         public void OnFinishCasting(Champion owner, Spell spell, Unit target) {
-
+            Champion champion = (Champion)target;
+            if (champion.Team != owner.Team){
+                spell.AddProjectileTarget("LuluWTwo", target);
+            } else {
```
Comment (molenzwiebel) [inline Champions/Lulu/W.cs]: Inconsistent indentation. The braces are on new lines everywhere else.

File: Champions/Lulu/W.cs [hunk 3]
```diff
@@ -15,16 +15,59 @@ public class W : GameScript
...
+                Particle p1 = ApiFunctionManager.AddParticleTarget(owner, "Lulu_W_buf_02.troy", target, 1);
```
Comment (molenzwiebel) [inline Champions/Lulu/W.cs]: p1 is never used.

File: Champions/Lulu/W.cs [hunk 4]
```diff
@@ -15,16 +15,59 @@ public class W : GameScript
...
-        }
+            Particle p = ApiFunctionManager.AddParticleTarget(owner, "Lulu_W_polymorph_01.troy", target, 1);
+            ApiFunctionManager.CreateTimer(time, () =>
+            {
+                ApiFunctionManager.RemoveParticle(p);
+                ApiFunctionManager.RemoveBuffHUDVisual(visualBuff);
+                owner.RemoveBuffGameScript(buff);
+                ApiFunctionManager.SetChampionModel((Champion)target, model);
+            });
+            projectile.setToRemove();
```
Comment (molenzwiebel) [inline Champions/Lulu/W.cs:57]: Lulu W gives movement speed if cast on an ally. I don't see that here.

**Review by molenzwiebel — CHANGES_REQUESTED** [review]

File: Champions/Lulu/W.cs (hunk 4 above)
Comment (molenzwiebel) [reply Champions/Lulu/W.cs:57]: Nevermind, git diffs hide information.

File: Buffs/LuluWBuff/LuluWBuff.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+{
+    class LuluWBuff : BuffGameScript
+    {
+        ChampionStatModifier statMod;
+
+        public void OnActivate(Unit unit, Spell ownerSpell)
+        {
+            statMod = new ChampionStatModifier();
+            statMod.MoveSpeed.PercentBonus = statMod.MoveSpeed.PercentBonus + 0.3f;
+            //statMod.MoveSpeed.PercentBonus = statMod.MoveSpeed.PercentBonus + this.GetStats().AbilityPower.Total * 0.3;
```
Comment (molenzwiebel) [inline Buffs/LuluWBuff/LuluWBuff.cs]: Remove this.

File: Champions/Lulu/W.cs [hunk 6]
```diff
@@ -15,16 +15,59 @@ public class W : GameScript
...
+         }
         public void OnUpdate(double diff) {
 
         }
-     }
+        void changeModel(float skinId, Unit target){
```
Comment (molenzwiebel) [inline Champions/Lulu/W.cs]: Why is this a float?

**Review by molenzwiebel — CHANGES_REQUESTED** [review]

File: Buffs/LuluWBuff/LuluWBuff.cs (hunk 5 above)
Comment (molenzwiebel) [reply Buffs/LuluWBuff/LuluWBuff.cs]: Actually, not remove, fix this. Adding 0.3 ms doesn't really do anything.

Comment (molenzwiebel) [reply Buffs/LuluWBuff/LuluWBuff.cs]: @deudly noted that this is 30% bonus. Even then, the implementation is incomplete.

File: Champions/Lulu/W.cs (hunk 1 above)
Comment (moonshadow565) [reply Champions/Lulu/W.cs]: Use a switch?

Comment (Deudly) [reply Champions/Lulu/W.cs]: Yeah @moonshadow565

File: Champions/Lulu/W.cs (hunk 3 above)
Comment (Deudly) [reply Champions/Lulu/W.cs]: kk

File: Buffs/LuluWBuff/LuluWBuff.cs (hunk 5 above)
Comment (Deudly) [reply Buffs/LuluWBuff/LuluWBuff.cs]: kk

File: Champions/Lulu/W.cs (hunk 6 above)
Comment (Deudly) [reply Champions/Lulu/W.cs]: kk

File: Champions/Lulu/W.cs (hunk 1 above)
Comment (Deudly) [reply Champions/Lulu/W.cs]: kk

File: Champions/Lulu/W.cs (hunk 4 above)
Comment (Deudly) [reply Champions/Lulu/W.cs:57]: kk

**Review by molenzwiebel — APPROVED** [review]

**Pull request merged**